Add more generic ProductIcon component that is going to replace PlanIcon (#37760)
Automattic/wp-calypso: client/components/product-icon/README.md

ProductIcon Component
=============

> ProductIcon component is a React component to display plans or products icons in SVG format.

Reviewer: ```suggestion
ProductIcon component is a React component to display plan or product icons in SVG format.
```

> We have four different plans icons:

Reviewer: Nit: It's generally more flexible if we just say that we have "the following" without specifying a number.

```suggestion
We have the following different plans icons:
```

- Free
- Personal
- Premium

> - Business

Reviewer: Minor: should we specify that Business is AKA "Professional"?
```suggestion
- Business (Professional)
```

Jetpack plans also use these plans icons.

> There is also one product icon:

Reviewer: Not specifying the number keeps this readme a bit more flexible:

```suggestion
There are also the following product icons:
```

- Jetpack Backup

## Usage

```jsx
import ProductIcon from 'components/product-icon';
import { PLAN_FREE, PLAN_BUSINESS } from 'lib/plans/constants';
import { PRODUCT_JETPACK_BACKUP_DAILY } from 'lib/products-values/constants';

export default function MyComponent() {
    return (
        <div className="my-component">
            <ProductIcon product={ PLAN_FREE } />
            <ProductIcon product={ PLAN_BUSINESS } />
            <ProductIcon product={ PRODUCT_JETPACK_BACKUP_DAILY } />
        </div>
    );
}

```

## Props

### `product`

Plan constant from `lib/plans/constants` or product constant from `lib/products-values/constants`. Can be one of:

- PLAN_FREE,
- PLAN_PERSONAL,
- PLAN_PREMIUM,

> - PLAN_BUSINESS,

Reviewer: Guess we don't have icons for WP.com blogger or ecommerce plans, is that intentional?

Author: Well, actually we have icons for both. I overlooked this error when moving the code over from `PlanIcon` component.

- PLAN_JETPACK_FREE,
- PLAN_JETPACK_BUSINESS,

> - PLAN_JETPACK_BUSINESS_MONTHLY,

Reviewer: I see we support monthly variants of Jetpack plans and Jetpack products, but we don't support 2-year plans for .com sites. Is that intentional?

Author: According to tests which are passing, we support 2-year plans. I'm going to update the readme file.

- PLAN_JETPACK_PREMIUM_MONTHLY,

> - PLAN_PERSONAL,

Reviewer 2: Lets move this higher up.
